### ISSUE TITLE
FR: only save generated file if it is different from existing file

### ISSUE THREAD

**Issue by gregveres** [frhagn/Typewriter#260]
HI @frhagn 

 I have now built up my project so that there are many many files being generated from the templates and I am using webpack's dev server to pack them up to be used during local testing. The end result is that when a TS file changes, my computer does a lot of work to get things built.

When I am working exclusively in C# on unit tests, what happens is that each time I save one of my C# files, it causes the template to be re-rendered and that causes the TS file to be written out, which causes the TS complier to run and webpack to repack the file. But all this work happens when nothing changed in the generated TS file. And what is worse is that the TS file that is getting generated is like a header file being included in many other TS files, causing a big chain reaction. 

I would like to see a feature in Typewriter where the generated file is created as a temp file, then the newly generated file is compared against the file that is already there from the last time the template ran. If the file is different, then the temp file is copied over the original file and everything works as it does today.
But if the temp file is exactly the same as the existing file, then just discard the tmp file. This will have the great side effect of not triggering all the other processes that run but don't need to run. 

Thanks
Greg

**Comment by agustinbcu01** [comment]
this is a work around. 
You can disable the auto generation.
![image](https://user-images.githubusercontent.com/24256131/38317875-a9fe71c6-37f3-11e8-9398-dfac0b973c68.png)

and generate only when you are sure

![image](https://user-images.githubusercontent.com/24256131/38318038-0f7c03ce-37f4-11e8-9800-92855399ae5f.png)

**Comment by gregveres** [comment]
Yes I agree that would work, but it isn’t a great solution. The tool should “do the right thing” without me having to drastically change my workflow. Me recognizing that I am about to not do typescript work for a while and turn off the generation is a pain. Even worse, it will lead to mistakes and wasted debugging time when I find later that I forgot to turn on the autogeneration and the files are out of sync. Typewriter is fantastic for keeping everything in sync - that is one of its strengths. I would hate to break that by turning it off all the time.
